crates/plugins/mediator-coordination/src/forward/routing.rs

```diff
@@ -0,0 +1,121 @@
+use std::fs;
+
+use didcomm::{Message, PackEncryptedOptions, UnpackOptions};
+
+use crate::web::{error::MediationError, AppState};
+
+struct MessageStore {
+    dirpath: String,
+}
+impl MessageStore {
+    fn persist_msg(&self, content: String, inode: String) {
+        fs::create_dir_all(&self.dirpath).unwrap();
+        let file = format!("{}/{}.json",&self.dirpath, inode,);
+        fs::write(file, content).unwrap();
+    }
+}
+
+
+pub async fn mediator_forward_process(
+    mediator_did: Option<&str>,
+    payload: &str,
+    state: &AppState,
+    store_dir_path: String,
+) -> Result<(), MediationError> {
+    // unpack encrypted payload message
+    let store = MessageStore { dirpath: store_dir_path };
+
+    let result = Message::unpack(
+        payload,
+        &state.did_resolver,
+        &state.secrets_resolver,
+        &UnpackOptions::default(),
+    )
+    .await;
+    {
+        match result {
+            Ok((unpack_msg, _)) => {
+                if unpack_msg.to.is_some() {
+                    let dids = Some(unpack_msg.clone().to).unwrap().unwrap();
+                    for did in dids {
+                        let (re_packed_msg, _) = unpack_msg
+                            .pack_encrypted(
+                                &did,
+                                mediator_did,
+                                None,
+                                &state.did_resolver,
+                                &state.secrets_resolver,
+                                &PackEncryptedOptions::default(),
+                            )
+                            .await
+                            .unwrap();
+
+                        store.persist_msg(serde_json::to_string_pretty(&re_packed_msg).unwrap(), did)
+                    }
+                }
+
+                Ok(())
+            }
+            Err(_) => Err(MediationError::MessageUnpackingFailure),
+        }
+    }
```

Comment:
You should avoid directly unwrapping code that has high chances of failures. instead use an error propagation mechanism to prevent crashes and facilitate debugging.
```rust
let (re_packed_msg, _) = unpack_msg
      .pack_encrypted(
       &did,
       mediator_did,
       None,
       &state.did_resolver,
       &state.secrets_resolver,
       &PackEncryptedOptions::default(),
)
.await
.map_err(|_| MediationError::MessagePackingFailure)?;
```
Also reduce nested conditions:
```rust
if let Some(dids) = unpack_msg.to {
    for did in dids {
        let (re_packed_msg, _) = unpack_msg
            .pack_encrypted(
                &did,
                mediator_did,
                None,
                &state.did_resolver,
                &state.secrets_resolver,
                &PackEncryptedOptions::default(),
            )
            .await
            .map_err(|_| MediationError::MessagePackingFailure)?;
        store.push(re_packed_msg);
    }
}
```